modules/simple-payments/simple-payments.php

```diff
@@ -286,6 +287,78 @@ function allow_sync_post_meta( $post_meta ) {
 		) );
 	}
 
+	/**
+	 * Enable Simple payments custom meta values for access through the REST API.
+	 * Field’s value will be exposed on a .meta key in the endpoint response,
+	 * and WordPress will handle setting up the callbacks for reading and writing
+	 * to that meta key.
+	 *
+	 * @link https://developer.wordpress.org/rest-api/extending-the-rest-api/modifying-responses/
+	 */
+	public function register_meta_fields_in_rest_api() {
+			register_meta( 'post', 'spay_price', array(
+				'description'       => esc_html__( 'Simple payments; price.', 'jetpack' ),
+				'object_subtype'    => self::$post_type_product,
+				'sanitize_callback' => 'absint',
```

Comment:
Noting that `absint()` will throw away any non-integer value from the price - is that intentional? I'd expect that we support two digits after decimal`0.00` for prices. We're probably doing this intentionally (as I see it's enforced in the UI as well), but I'd love to know why.

Reply:
Good point — there's some confusion about price since the current button in Calypso complains about using digits but placeholder text suggests they're ok.

@rodrigoi can you elaborate on spec of these values a bit, especially the price has been difficult to navigate.

Reply:
This now has a proper price and currency validation.

I'm not validating differently based on different currencies since that seems quite tricky to do here. (Three currencies don't allow decimals). We are validating for that in the client and PayPal will throw an error when attempting to use it with such price.

Reply:
I'll mark this as _resolved_ because, as @simison pointed out, as long as the block UI has some validation, the payment processor will catch those and report them as errors.